### PR TITLE
Add linter for the JavaScript port

### LINKED ISSUE #16: JavaScript: Add eslint
For the JavaScript port of the package present in the repo, we should add eslint as a dev dependency to ensure consistent code quality.

### PR DESCRIPTION
**Pull request by sudiptog81**
### Summary

* Added a linter for the JS port to enforce code style and code consistency.
* Added scripts to check and fix linter issues.

Closes #16.

### CODE REVIEW

File: javascript/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            "never"
+        ],
+        "valid-jsdoc": "error",
+        "vars-on-top": "error",
+        "wrap-iife": "error",
+        "wrap-regex": "error",
+        "yield-star-spacing": "error",
+        "yoda": "error"
+    }
+}
```
Comment (MattIPv4) [inline javascript/.eslintrc.json]:
```suggestion
}

```
newline eof

File: javascript/package.json [hunk 2]
```diff
@@ -9,6 +9,13 @@
...
+    "lint:fix": "eslint lib/**/*.js --fix"
+  },
   "author": "DigitalOcean, Matt Cowley",
-  "license": "Apache-2.0"
-}
+  "license": "Apache-2.0",
+  "devDependencies": {
+    "eslint": "^7.12.0"
+  }
+}
```
Comment (MattIPv4) [inline javascript/package.json]:
```suggestion
}

```
newline eof

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: javascript/.eslintrc.json (hunk 1 above)
Comment (sudiptog81) [reply javascript/.eslintrc.json]: VSCode :)

**Review by MattIPv4 — COMMENTED** [review]
This seems to be setting a huge number of custom rules beyond the default recommended -- are they all needed, I'd prefer to keep our customisations minimal if possible initially?

**Comment by sudiptog81** [comment]
I chose the option to inspect the actual file to detect the code style, however, we can definitely keep the configuration minimal by removing the rules key. I'll make the changes.

File: javascript/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+    "env": {
+        "browser": true,
+        "commonjs": true,
+        "es6": true,
+        "node": true
```
Comment (MattIPv4) [inline javascript/.eslintrc.json]:
```suggestion
        "commonjs": true,
        "es6": true
```
We don't need node-specific or browser-specific features at present.

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
I'm happy to set some basic rules based on the code style present, but what was present before seemed to be setting absolutely everything, which is rather excessive imo.

I think just setting all the whitespace specific stuff that isn't default would be best, as that'll do the best job of enforcing consistent code -- we can add more rules as they come up in code changes etc.:
```json
    "rules": {
        "space-before-function-paren": [
            "error",
            {
                "anonymous": "always",
                "named": "never",
                "asyncArrow": "always"
            }
        ],
        "object-curly-spacing": [
            "error",
            "always"
        ],
        "prefer-const": "error",
        "indent": [
            "error",
            4,
            {
                "SwitchCase": 1
            }
        ],
        "semi": [
            "error",
            "always"
        ],
        "quotes": [
            "error",
            "single",
            {
                "avoidEscape": true
            }
        ],
        "object-curly-newline": [
            "error",
            {
                "multiline": true,
                "consistent": true
            }
        ],
        "comma-dangle": [
            "error",
            "always-multiline"
        ],
        "comma-spacing": [
            "error",
            {
                "before": false,
                "after": true
            }
        ],
        "comma-style": [
            "error",
            "last"
        ],
        "eol-last": "error",
        "key-spacing": [
            "error",
            {
                "beforeColon": false,
                "afterColon": true
            }
        ],
        "keyword-spacing": [
            "error",
            {
                "before": true,
                "after": true
            }
        ],
        "block-spacing": "error",
        "space-in-parens": [
            "error",
            "never"
        ],
        "space-before-blocks": "error",
        "no-trailing-spaces": "error",
        "semi-spacing": [
            "error",
            {
                "before": false,
                "after": true
            }
        ],
        "space-infix-ops": "error",
        "no-empty": "error",
        "max-len": [
            "error",
            {
                "code": 120,
                "ignoreComments": true,
                "ignoreStrings": true,
                "ignoreTemplateLiterals": true,
                "ignoreRegExpLiterals": true
            }
        ]
    }
```

**Review by MattIPv4 — APPROVED** [review]
Lgtm, tyvm!

**Pull request merged**